src/yaml/from_yaml.cr

```diff
@@ -21,16 +21,23 @@ def Array.from_yaml(string_or_io)
 end
 
 def Nil.new(pull : YAML::PullParser)
-  value = pull.read_scalar
-  if value.empty?
+  case value = pull.read_scalar
+  when "", "NULL", "Null", "null", "~"
     nil
   else
     raise YAML::ParseException.new("Expected nil, not #{value}", 0, 0)
   end
 end
 
 def Bool.new(pull : YAML::PullParser)
-  pull.read_scalar == "true"
+  case value = pull.read_scalar
+  when "true", "True", "TRUE", "on", "On", "ON", "y", "Y", "yes", "Yes", "YES"
```

Comment:
These values are from the Language-Independent Types for YAML. see [my comment above](https://github.com/crystal-lang/crystal/pull/4838#issuecomment-323063279).